linkml_runtime/utils/jsonasobj2/_jsonobj.py

```diff
@@ -0,0 +1,315 @@
+import json
+from typing import Union, List, Dict, Tuple, Optional, Callable, Any, Iterator
+from hbreader import hbread
+import warnings
+from .extendednamespace import ExtendedNamespace
+
+# Possible types in the JsonObj representation
+JsonObjTypes = Union["JsonObj", List["JsonObjTypes"], str, bool, int, float, None]
+
+# Types in the pure JSON representation
+JsonTypes = Union[Dict[str, "JsonTypes"], List["JsonTypes"], str, bool, int, float, None]
+
+# Control variables -- note that subclasses can add to this list
+hide = ['_if_missing', '_root']
+
+# https://github.com/linkml/linkml/issues/1966
+warnings.warn("This module is going to be replaced by native Python dataclasses.", PendingDeprecationWarning)
```

Comment:
separate note on this, I think we will want to give some instructions on what to do here - we are warning someone that something will happen, but don't tell them what to do about it (so then the only way they can silence the warning is to filter it, but then they might miss if it gets updated with instructions). Are the instructions we want to give here just "replace jsonasobj2 stuff with equivalent dataclasses stuff"? if so i think we'll probably want to write an update guide for the docs.

@cmungall we might want to give a version warning too, like "jsonasobj2 will be removed in x version" so people can either set version caps or know when they need to update by, any input on what that version should be?